firmware/libraries/versavis/src/ADIS16445.cpp

```diff
@@ -373,20 +373,31 @@ float ADIS16445::tempScale(int16_t sensorData) {
 // Method to update the sensor data without any validity checks (may result in
 // spikes).
 ///////////////////////////////////////////////////////////////////////////////////////////////
-void ADIS16445::updateData() { sensor_data_ = sensorReadAll(); }
+bool ADIS16445::updateData() {
+  sensor_data_ = sensorReadAll();
+  return true;
+}
 
 /////////////////////////////////////////////////////////////////////////////////////////////
 // Method to update the internally stored sensor data recusivelly by checking
 // the validity.
 ///////////////////////////////////////////////////////////////////////////////////////////////
-void ADIS16445::updateDataRecursive(unsigned int depth) {
+bool ADIS16445::updateDataRecursive(const unsigned int depth, bool success) {
```

Comment:
First, could you transform this to a tail recursion for better performance?  Secondly, shouldn't success be updated with a logical &. (The former would also solve the latter)